src/charm.py

```diff
@@ -96,18 +94,60 @@ def _are_relations_ready(self) -> bool:
         return self._tls.get_tls_relation() is not None
 
     def _reconcile(self) -> None:
-        """Reconcile charm status based on configuration and integrations."""
+        """Reconcile charm status based on configuration and integrations.
+
+        Raises:
+            RuntimeError: when initializing the lightkube client fails,
+            or when creating the gateway resource fails.
+        """
+        try:
```

Comment:
Are you able to check if the charm was deployed using `--trust` via Juju or Ops ?